### PR TITLE
#113 Allow defining multiple Token Info endpoints as fallback

### PR DESCRIPTION
**Pull request by vroldanbet**
- Fallback should will also happen when tokeninfo responds with payload "error"

### CODE REVIEW

**Comment by vroldanbet** [comment]
@hjacobs I missed checking when first url returns "error" body, which is the normal scenario.

File: src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java [hunk 1]
```diff
@@ -1,45 +1,85 @@
...
 
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class FallbackTokenInfoRequestExecutorTest extends AbstractOAuthTest {
 
-    private Integer wireMockPort = Integer.valueOf(System.getProperty("wiremock.port", "10080"));
+    @Rule
+    public WireMockRule secondServerMock = new WireMockRule(10081);
```
Comment (hjacobs) [inline src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java]: port 10081 here...

File: src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java [hunk 2]
```diff
@@ -1,45 +1,85 @@
...
 
     @Test
     public void testSingleEndpointValidToken() {
         FallbackTokenInfoRequestExecutor executor =
-                new FallbackTokenInfoRequestExecutor("http://localhost:" + wireMockPort + "/tokeninfo");
+                new FallbackTokenInfoRequestExecutor("http://localhost:10080/tokeninfo");
```
Comment (hjacobs) [inline src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java]: and hardcoded 10080 here?

File: src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java [hunk 3]
```diff
@@ -1,45 +1,85 @@
...
         assertThat(result).hasSize(7);
     }
 
     @Test(expected = HttpClientErrorException.class)
-    public void testSingleEndpointInvalidToken() {
+    public void testSingleEndpointNotFound() {
         FallbackTokenInfoRequestExecutor executor =
-                new FallbackTokenInfoRequestExecutor("http://localhost:" + wireMockPort + "/tokeninfo");
-        executor.getMap(INVALID_ACCESS_TOKEN);
+                new FallbackTokenInfoRequestExecutor("http://localhost:10080/tokeninfo");
```
Comment (hjacobs) [inline src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java]: maybe put the /tokeninfo URLs in some constants, it's a lot of repeated code and hard to read (with this magic port number)

File: src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java (hunk 2 above)
Comment (vroldanbet) [inline src/test/java/org/zalando/planb/revocation/config/FallbackTokenInfoRequestExecutorTest.java]: Yeah, was trying to keep it simple, but its true that's not the case if the environment is set to a different value.

**Comment by vroldanbet** [comment]
@hjacobs done

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**